tools/bin/ecflow/processes/main/meteorological/get_BC_metdata.py

```diff
@@ -0,0 +1,143 @@
+
+from tonic.io import read_config
+from monitor import model_tools
+
+
+def main():
+    # read in configuration file
+    parser = argparse.ArgumentParser(description='Download met data for ' +
+                                     'British Columbia')
+    parser.add_argument('config_file', metavar='config_file',
+                        help='configuration file')
+    args = parser.parse_args()
+    config_dict = read_config(args.config_file)
+
+    # get units conversion from cf_units for K to degC
+    units_in = cf_units.Unit('K')
+    units_out = cf_units.Unit('degC')
+
+    # initial cdo
+    cdo = Cdo()
+
+    # read in meteorological data location
+    old_config_file = config_dict['ECFLOW']['old_Config']
+    new_config_file = config_dict['ECFLOW']['new_Config']
+    n_days = int(config_dict['ECFLOW']['Met_Delay'])
+    met_out = config_dict['MONITOR']['Orig_Met']
+
+    # read in the source and destination paths and current date
+    source_loc = config_dict['MONITOR']['BC_Met_Source']
+    dest_loc = os.path.dirname(met_out)
+
+    # read in grid_file from config file
+    grid_file = config_dict['DOMAIN']['GridFile']
+
+    # get current date and number of days
+    # define the end date for subdaily generation and the vic run
+    end_date = datetime.now() - timedelta(days=n_days)
+    end_date_format = end_date.strftime('%Y-%m-%d')
+
+    # get VIC start date and the date to save the state
+    # the last 60 days in the met dataset are provisional, so we start our run
+    # the day before the first provisional day
+    start_date = end_date - timedelta(days=60)
+    start_date_format = start_date.strftime('%Y-%m-%d')
+    # we save the state at the beginning of the first provisional day
+    vic_save_state = start_date + timedelta(days=1)
+    vic_save_state_format = vic_save_state.strftime('%Y-%m-%d')
+
+
+    # check if data we are downloading coms from multiple years
+    # replace start date, end date and met location in the configuration file
+    kwargs = {'START_DATE': start_date_format,
+              'END_DATE': end_date_format,
+              'VIC_SAVE_STATE': vic_save_state_format}
+    model_tools.replace_var_pythonic_config(old_config_file, new_config_file,
+                                            header=None, **kwargs)
+    # an abbreviation and a full name is needed
+    varnames = [('pr', 'precipitation_amount'), ('tmmn', 'air_temperature'),
+                ('tmmx', 'air_temperature'), ('vs', 'wind_speed'),
+                ('srad', 'surface_downwelling_shortwave_flux_in_air'),
+                ('sph', 'specific_humidity')]
+    # set up a connection to the NKN network
+    ssh = paramiko.SSHClient()
+    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    ssh.load_host_keys(os.path.expanduser(
+        os.path.join("~", ".ssh", "known_hosts")))
+    ssh.connect('reacchdb.nkn.uidaho.edu', username='vicmet', password='cl!m@te')
+    sftp = ssh.open_sftp()
+    # access data files
```

Comment:
I haven't used `paramiko` before so maybe it's already built into this and I just don't realize it, but I could envision a situation where paramiko tries to set up a connection to the NKN network and it fails because the NKN network isn't working or there isn't a connection to the internet from where this is being run for some reason. So perhaps add a useful error message if it is unable to establish a connection (if paramiko doesn't do that already)

Reply:
`paramiko` has pretty good error messaging that covers this circumstance.